internal/docstore/dynamodocstore/query.go

```diff
@@ -166,12 +167,20 @@ func (c *collection) bestQueryable(q *driver.Query) (indexName *string, pkey, sk
 	return nil, "", ""
 }
 
-// Reports whether the fields selected by the query are projected into (that is,
-// contained directly in) the global index. We need this check before using the
-// index, because if a global index doesn't have all the desired fields, then a
-// separate RPC for each returned item would be necessary to retrieve those fields,
-// and we'd rather scan than do that.
-func (c *collection) fieldsIncluded(q *driver.Query, gi *dynamodb.GlobalSecondaryIndexDescription) bool {
+// localFieldsIncluded reports whether a local index supports all the selected fields
+// of a query. Since DynamoDB will read explicitly provided fields from the table if
+// they are not projected into the index, the only case where a local index cannot
+// be used is when the query wants all the fields, and the index projection is not ALL.
```

Comment:
Could you add a link of where this is documented, I'm talking about the last sentence.

Reply:
Done.